### PR TITLE
Update change log due to Realm Core update

### PR DESCRIPTION
**Pull request by kneth**
@bmunkholm suggests in #1030 that we mention Realm Core updates in the change log.

@bmunkholm @emanuelez @cmelchior

### CODE REVIEW

**Comment by cmelchior** [comment]
I would probably specify that it was updated to 0.89, but no major issue if not.
:+1:

File: changelog.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@
  * Added Realm.deleteRealmFile(File) method.
  * Fixed bug causing queries to fail if multiple Realms has different field ordering.
  * Fixed bug when using Realm.copyToRealm() with a primary key could crash if default value was already used in the Realm.
+ * Realm Core updated: performance improvements, bug fixes related to string based primary keys.
```
Comment (emanuelez) [inline changelog.txt]: To keep persistency in the wording I would suggest something like:
Updated Ream Core: performance improvements, bug fixes related to string based primary keys.

**Comment by bmunkholm** [comment]
I would be more specific than this one liner summary. See Cocoa.

**Comment by kneth** [comment]
Please rereview @bmunkholm @cmelchior @emanuelez 

As something new, updating to a new Realm Core version, the change log now explains the impact on the binding. I haven't any benchmarks for the performance improvements but according to our Cocoa binding, they are "significant".

**Comment by cmelchior** [comment]
:+1:

**Comment by emanuelez** [comment]
:+1:

**Pull request merged**